src/windowing.jl

```diff
@@ -0,0 +1,61 @@
+struct WindowViewer{T<:Real, V<:AbstractVector{<:T}}
+    timeseries::V
+    halfwidth::Int
+    stride::Int
+    strided_indices::AbstractVector{Int}
+end
+
+"""
+
+    WindowViewer(
+        timeseries::V,
+        halfwidth::Int,
+        stride::Int,
+    ) where {T<:Real, V<:AbstractVector{<:T}}
+
+Returns struct of WindowViewer with fields:
+- `WindowViewer.timeseries`
+- `WindowViewer.halfwidth`
+- `WindowViewer.stride`
+- `WindowViewer.strided_indices`: a vector of indices at which the window viewing should be applied.
+"""
+function WindowViewer(
+    timeseries::V,
+    halfwidth::Int,
+    stride::Int,
+) where {T<:Real, V<:AbstractVector{<:T}}
+
+    n = length(timeseries)
+    strided_indices = get_stride_indices(n, halfwidth, stride)
+    return WindowViewer(timeseries, halfwidth, stride, strided_indices)
+end
+
+"""
+
+    bracket(idx::Int, wv::WindowViewer)
+
+Returns index span based on target index and parameters of the window viewer.
+"""
+bracket(idx::Int, wv::WindowViewer) = (idx-2*wv.halfwidth, idx)
```

Comment:
I would remoe this function and instead use ` (idx-2*wv.halfwidth, idx)` directly in the extension of `base.iterate`